### PR TITLE
Trac 875

### PR DESCRIPTION
**Pull request by cdeaneGit**
This is a PULL request in support of JIRA TRAC-875.

New code from utk-fedora-fedoragsearch-solr-config(TRAC-875) should be merged to
utk-fedora-fedoragsearch-solr-config(institutional_repository)

Code changes are in /utk-fedora-fedoragsearch-solr-config/fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt

Testing instructions:

    vagrant destroy --force
    git pull 
    git checkout -b remotes/origin/TRAC-875
    vagrant up
    -- TRACE(TRAC-875) will load the utk-fedora-fedoragsearch-solr-config(TRAC-875)

    userA: ingest one thesis. 
    thesis_manager: accept-publish
    admin: Check the url of the thesis to see the pid
    For example: http://localhost:8000/islandora/object/utk.ir.td%3A16
    If you are logged in as admin you can also go the the datastreams page.
    This item has PID  utk.ir.td:16

    Update the solr index so that you can find utk.ir.td:16
       http://localhost:8080/fedoragsearch/rest?operation=updateIndex

    List all pids in solr index on localhost
    http://localhost:8080/solr/collection1/select?q=*%3A*&sort=PID+asc&rows=1000000&fl=PID&wt=csv&indent=true

    Look at the data display at:
    http://localhost:8080/solr/collection1/select?q=PID%3A%22utk.ir.td%3A16%22&fl=utk_mods_*&wt=xml&indent=true

    Look at the at the solr schema browser at: http://localhost:8080/solr/#/collection1/schema-browser
    See if you can locate the fieldname: utk_mods_etd_author_s

@markpbaggett @CanOfBees

### CODE REVIEW

**Comment by markpbaggett** [comment]
@cdeaneGit reviewing now.   Removed file READ_TRAC-875.txt

**Comment by markpbaggett** [comment]
@cdeaneGit , based on results, I'm not sure this does exactly what it's supposed to do. The Solr field is right, but the serialization should follow the meeting plan notes:

https://wiki.lib.utk.edu/pages/viewpage.action?pageId=39387444

Format as Family, Given, termsOfAddress

I'll be more explicit in my review.

File: READ_TRAC-875.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+Define utk_mods_etd_author in Solr 
+
+This uses TRACE (Trac-875) branch 
+with utk-fedora-fedoragsearch-solr-config (TRAC-875) branch.
+
+My changes will be made in utk-fedora-fedoragsearch-solr-config (TRAC-875) branch.
+
+Then both branches will have to be pushed to origin for their branches.
+
+
```
Comment (markpbaggett) [inline READ_TRAC-875.txt]: This whole file needs to be dropped from tracking. It's tied to a commit or series of commits; not code.

From the command line do:

git checkout TRAC-875
git rm READ_TRAC_875.txt
git push origin TRAC-875

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 2]
```diff
@@ -55,6 +56,57 @@
...
+  <xsl:template match="mods:mods/mods:name[(mods:role/mods:roleTerm='Author') or 
+    (mods:role/mods:roleTerm='author')]" mode="utk_mods_etd">
+    <xsl:variable name="given-n" select="mods:namePart[@type='given']"/>
+    <xsl:variable name="family-n" select="mods:namePart[@type='family']"/>
+    <xsl:variable name="t-o-address" select="mods:namePart[@type='termsOfAddress']"/>
+
+     <field name="utk_etd_mods_name_author_s">
+       <xsl:choose>
+          <xsl:when test="$t-o-address!=''">
+               <xsl:value-of select="concat($given-n, ' ', $family-n, ', ', $t-o-address)"/>
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: This is wrong.  It should be:

<xsl:value-of select="concat($family-n, ', ', $given-n, ', ', $t-o-address)"/>

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 3]
```diff
@@ -55,6 +56,57 @@
...
+        </xsl:when>
+        <xsl:otherwise>
+              <xsl:value-of select="concat($given-n, ' ', $family-n)"/>
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: This is wrong, it should be:



<xsl:value-of select="concat($family-n, ', ', $given-n)"/>

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 4]
```diff
@@ -55,6 +56,57 @@
...
+  </xsl:template>
+
+
+  <!-- utk_mods_etd  -->
+  <!-- utk_mods_etd  -->
+
+  
+  <!-- JIRA TRAC-875 Define utk_mods_etd_author in Solr -->
+  <!-- the following template creates an _s field for single etd author -->
+  <xsl:template match="mods:mods/mods:name[(mods:role/mods:roleTerm='Author') or (mods:role/mods:roleTerm='author')]" mode="utk_ir_MODS">
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Do we have data where the roleTerm is lower case?

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 5]
```diff
@@ -55,6 +56,57 @@
...
+    <xsl:variable name="given-n" select="mods:namePart[@type='given']"/>
+    <xsl:variable name="family-n" select="mods:namePart[@type='family']"/>
+    <xsl:variable name="t-o-address" select="mods:namePart[@type='termsOfAddress']"/>
+
+    <field name="utk_mods_etd_author_s">
+      <xsl:choose>
+        <xsl:when test="$t-o-address!=''">
+          <xsl:value-of select="concat($given-n, ' ', $family-n, ', ', $t-o-address)"/>
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Same as above.

<xsl:value-of select="concat($family-n, ', ', $given-n, ', ', $t-o-address)"/>

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 6]
```diff
@@ -55,6 +56,57 @@
...
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="concat($given-n, ' ', $family-n)"/>
```
Comment (markpbaggett) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Same as above.

<xsl:value-of select="concat($family-n, ', ', $given-n)"/>

**Comment by markpbaggett** [comment]
@cdeaneGit I think my review is done.  Please respond to or address directly any of the suggestions.  If you'd like, also ask @CanOfBees about his thoughts.

**Review by markpbaggett — CHANGES_REQUESTED** [review]

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 4 above)
Comment (CanOfBees) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Let's not worry about this one.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 7]
```diff
@@ -34,6 +34,7 @@
       mods:genre = 'Academic theses'.
     -->
     <xsl:apply-templates mode="utk_ir_MODS" select="$content//mods:mods[1]"/>
+	<xsl:apply-templates mode="utk_mods_etd" select="$content//mods:mods[1]"/>
```
Comment (CanOfBees) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Let's drop this. We don't need another mode at this point.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 8]
```diff
@@ -43,7 +44,7 @@
     <xsl:variable name="family-n" select="mods:namePart[@type='family']"/>
     <xsl:variable name="t-o-address" select="mods:namePart[@type='termsOfAddress']"/>
 
-    <field name="utk_mods_etd_name_author_ms">
+    <field name="utk_mods_ir_name_author_ms">
```
Comment (CanOfBees) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Switch field name to utk_mods_etd_name_author_ms

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 9]
```diff
@@ -55,6 +56,57 @@
...
+  <!-- JIRA TRAC-875 Define utk_mods_etd_author in Solr -->
+  <!-- the following template creates an _s field for single etd author -->
+  <!-- need to  use single_valued_hashset -->
+  <xsl:template match="mods:mods/mods:name[(mods:role/mods:roleTerm='Author') or 
+    (mods:role/mods:roleTerm='author')]" mode="utk_mods_etd">
+    <xsl:variable name="given-n" select="mods:namePart[@type='given']"/>
+    <xsl:variable name="family-n" select="mods:namePart[@type='family']"/>
+    <xsl:variable name="t-o-address" select="mods:namePart[@type='termsOfAddress']"/>
+
+     <field name="utk_etd_mods_name_author_s">
```
Comment (CanOfBees) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: `utk_mods_etd_name_...`

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 10]
```diff
@@ -55,6 +56,57 @@
...
+        </xsl:when>
+        <xsl:otherwise>
+              <xsl:value-of select="concat($given-n, ' ', $family-n)"/>
+        </xsl:otherwise>
+        </xsl:choose>
+    </field>
+  </xsl:template>
+
+
+  <!-- utk_mods_etd  -->
```
Comment (CanOfBees) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: delete lines 85 through 108, they're unnecessary.

**Review by CanOfBees — CHANGES_REQUESTED** [review]

File: READ_TRAC-875.txt (hunk 1 above)
Comment (cdeaneGit) [reply READ_TRAC-875.txt]: Done

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 2 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Done.  Added missing comma.
I am pushing each little change with a descriptive commit message.

Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Done. Fixed display order to "family, given, t-o-address" on 2 lines.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 3 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Done. Fixed display order to "family, given, t-o-address" on 2 lines.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 4 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Not addressing this one.

What I am addressng is REMOVE second copy  of the JIRA TRAC-875 template that creates the field utk_etd_mods_author_s.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 5 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Done by removing second copy of the template for utk_mods_etd_author_s.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 7 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Done.  Remove mode="utk_mods_etd"  line.

Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Also removed extra template just above my target template in the code.  Brider looked at the file and suggested it..  About 16 lines.

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 8 above)
Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Done

**Comment by markpbaggett** [comment]
@cdeaneGit taking a final look now.

**Comment by markpbaggett** [comment]
@cdeaneGit this looks good to me.  The one thing I want to ask is why this only gets an ms and an mt.  Can you ask @CanOfBees this?

**Review by markpbaggett — APPROVED** [review]

**Comment by CanOfBees** [comment]
@cdeaneGit taking a look at this now

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt [hunk 11]
```diff
@@ -34,77 +34,35 @@
...
   <!-- need to  use single_valued_hashset -->
   <xsl:template match="mods:mods/mods:name[(mods:role/mods:roleTerm='Author') or 
-    (mods:role/mods:roleTerm='author')]" mode="utk_mods_etd">
+    (mods:role/mods:roleTerm='author')]" mode="utk_ir_MODS">
     <xsl:variable name="given-n" select="mods:namePart[@type='given']"/>
     <xsl:variable name="family-n" select="mods:namePart[@type='family']"/>
     <xsl:variable name="t-o-address" select="mods:namePart[@type='termsOfAddress']"/>
 
-     <field name="utk_etd_mods_name_author_s">
+     <field name="utk_mods_name_author_ms">
```
Comment (CanOfBees) [inline fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: @cdeaneGit everything looks great except for this one field: please add '_etd_' to the field name; e.g. `utk_mods_etd_name_author_ms`.

**Review by CanOfBees — CHANGES_REQUESTED** [review]

File: fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt (hunk 11 above)
Comment (markpbaggett) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Sorry for missing that.  Yes, add those 4 characters, add, commit, and push so we can merge.

Comment (cdeaneGit) [reply fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt]: Done.  utk_mods_etd_name_author_ms

**Comment by cdeaneGit** [comment]
Done.  Revise fieldname to utk_mods_etd_name_author_ms

**Comment by markpbaggett** [comment]
@cdeaneGit, that looks right to me.  @CanOfBees if you agree, approve review and I will merge.

**Review by CanOfBees — APPROVED** [review]
@cdeaneGit looks great - thank you for making the change.

**Pull request merged**